can't not build at aarch64

I have project need to build for aarch64 Architecture.
when build rust-rdkafka-0.20. I got some error.
```
 --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/client.rs:120:76
    |
120 |         unsafe { rdsys::rd_kafka_conf_set_log_cb(native_config.ptr(), Some(native_log_cb::<C>)) };
    |                                                                            ^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `unsafe extern "C" fn(*const rdsys::rd_kafka_s, i32, *const u8, *const u8)`
               found type `unsafe extern "C" fn(*const rdsys::rd_kafka_s, i32, *const i8, *const i8) {client::native_log_cb::<C>}`

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/client.rs:121:78
    |
121 |         unsafe { rdsys::rd_kafka_conf_set_stats_cb(native_config.ptr(), Some(native_stats_cb::<C>)) };
    |                                                                              ^^^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `unsafe extern "C" fn(*mut rdsys::rd_kafka_s, *mut u8, usize, *mut std::ffi::c_void) -> i32`
               found type `unsafe extern "C" fn(*mut rdsys::rd_kafka_s, *mut i8, usize, *mut std::ffi::c_void) -> i32 {client::native_stats_cb::<C>}`

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/client.rs:122:78
    |
122 |         unsafe { rdsys::rd_kafka_conf_set_error_cb(native_config.ptr(), Some(native_error_cb::<C>)) };
    |                                                                              ^^^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `unsafe extern "C" fn(*mut rdsys::rd_kafka_s, i32, *const u8, *mut std::ffi::c_void)`
               found type `unsafe extern "C" fn(*mut rdsys::rd_kafka_s, i32, *const i8, *mut std::ffi::c_void) {client::native_error_cb::<C>}`

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/client.rs:125:74
    |
125 |             rdsys::rd_kafka_new(rd_kafka_type, native_config.ptr_move(), errstr.as_ptr() as *mut i8, errstr.len())
    |                                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*mut u8`
               found type `*mut i8`

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/client.rs:274:30
    |
274 |     let fac = CStr::from_ptr(fac).to_string_lossy();
    |                              ^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/client.rs:275:38
    |
275 |     let log_message = CStr::from_ptr(buf).to_string_lossy();
    |                                      ^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/client.rs:309:33
    |
309 |     let reason = CStr::from_ptr(reason).to_string_lossy();
    |                                 ^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/config.rs:154:44
    |
154 |                                            errstr.as_ptr() as *mut i8, errstr.len())
    |                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*mut u8`
               found type `*mut i8`

error[E0308]: mismatched types
  --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/consumer/mod.rs:54:52
   |
54 |                 let error = unsafe { cstr_to_owned(rdsys::rd_kafka_err2str(err)) };
   |                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i8, found u8
   |
   = note: expected type `*const i8`
              found type `*const u8`

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/consumer/base_consumer.rs:179:48
    |
179 |             let error = unsafe { cstr_to_owned(rdsys::rd_kafka_err2str(ret_code)) };
    |                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i8, found u8
    |
    = note: expected type `*const i8`
               found type `*const u8`

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/consumer/base_consumer.rs:192:48
    |
192 |             let error = unsafe { cstr_to_owned(rdsys::rd_kafka_err2str(ret_code)) };
    |                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i8, found u8
    |
    = note: expected type `*const i8`
               found type `*const u8`

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/util.rs:106:20
    |
106 |     CStr::from_ptr(bytes_cstr.as_ptr()).to_string_lossy().into_owned()
    |                    ^^^^^^^^^^^^^^^^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`

error[E0308]: mismatched types
   --> /root/.cargo/registry/src/mirrors.ustc.edu.cn-b63e9dae659fc205/rdkafka-0.20.0/src/util.rs:111:20
    |
111 |     CStr::from_ptr(cstr).to_string_lossy().into_owned()
    |                    ^^^^ expected u8, found i8
    |
    = note: expected type `*const u8`
               found type `*const i8`

error: aborting due to 13 previous errors

```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,20 +1,25 @@
 language: rust
 
-rust:
-  - stable
-  - nightly-2019-10-17
-
-sudo: required
-
 services:
   - docker
 
-os:
-  - linux
-  - osx
+jobs:
+  include:
+  - os: linux
+    arch: amd64
+    rust: stable
+  - os: linux
+    arch: amd64
+    rust: nightly-2019-10-17
+    env: RDKAFKA_RUN_TESTS=1
+  - os: linux
+    arch: arm64
+    rust: stable
+  - os: osx
+    rust: stable
 
 script:
-  - if [[ "$TRAVIS_OS_NAME" == "linux" && "$TRAVIS_RUST_VERSION" == nightly*  ]]; then ./test_suite.sh; else cargo build --verbose; fi
+  - if [[ "$RDKAFKA_RUN_TESTS" ]]; then ./test_suite.sh; else cargo build --verbose; fi
 
 notifications:
   webhooks:

--- a/src/consumer/base_consumer.rs
+++ b/src/consumer/base_consumer.rs
@@ -176,7 +176,7 @@ impl<C: ConsumerContext> Consumer<C> for BaseConsumer<C> {
         }
         let ret_code = unsafe { rdsys::rd_kafka_subscribe(self.client.native_ptr(), tpl.ptr()) };
         if ret_code.is_error() {
-            let error = unsafe { cstr_to_owned(rdsys::rd_kafka_err2str(ret_code) as *const i8) };
+            let error = unsafe { cstr_to_owned(rdsys::rd_kafka_err2str(ret_code)) };
             return Err(KafkaError::Subscription(error));
         };
         Ok(())
@@ -189,7 +189,7 @@ impl<C: ConsumerContext> Consumer<C> for BaseConsumer<C> {
     fn assign(&self, assignment: &TopicPartitionList) -> KafkaResult<()> {
         let ret_code = unsafe { rdsys::rd_kafka_assign(self.client.native_ptr(), assignment.ptr()) };
         if ret_code.is_error() {
-            let error = unsafe { cstr_to_owned(rdsys::rd_kafka_err2str(ret_code) as *const i8) };
+            let error = unsafe { cstr_to_owned(rdsys::rd_kafka_err2str(ret_code)) };
             return Err(KafkaError::Subscription(error));
         };
         Ok(())

--- a/src/consumer/mod.rs
+++ b/src/consumer/mod.rs
@@ -51,7 +51,7 @@ pub trait ConsumerContext: ClientContext {
             }
             RDKafkaRespErr::RD_KAFKA_RESP_ERR__REVOKE_PARTITIONS => Rebalance::Revoke,
             _ => {
-                let error = unsafe { cstr_to_owned(rdsys::rd_kafka_err2str(err) as *const i8) };
+                let error = unsafe { cstr_to_owned(rdsys::rd_kafka_err2str(err)) };
                 error!("Error rebalancing: {}", error);
                 Rebalance::Error(error)
             }

--- a/src/util.rs
+++ b/src/util.rs
@@ -102,12 +102,12 @@ impl<T: Send + Sync> IntoOpaque for Box<T> {
 
 // TODO: check if the implementation returns a copy of the data and update the documentation
 /// Converts a byte array representing a C string into a String.
-pub unsafe fn bytes_cstr_to_owned(bytes_cstr: &[i8]) -> String {
+pub unsafe fn bytes_cstr_to_owned(bytes_cstr: &[c_char]) -> String {
     CStr::from_ptr(bytes_cstr.as_ptr() as *const c_char).to_string_lossy().into_owned()
 }
 
 /// Converts a C string into a String.
-pub unsafe fn cstr_to_owned(cstr: *const i8) -> String {
+pub unsafe fn cstr_to_owned(cstr: *const c_char) -> String {
     CStr::from_ptr(cstr as *const c_char).to_string_lossy().into_owned()
 }
 
@@ -122,7 +122,7 @@ impl ErrBuf {
         ErrBuf { buf: [0; ErrBuf::MAX_ERR_LEN] }
     }
 
-    pub fn as_mut_ptr(&mut self) -> *mut i8 {
+    pub fn as_mut_ptr(&mut self) -> *mut c_char {
         self.buf.as_mut_ptr()
     }
 

